dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3832,18 +3832,27 @@ def test_metadata_task_size(tmpdir, engine, write_metadata_file, metadata_task_s
     assert_eq(ddf2b, ddf2c)
 
 
-def test_extra_file(tmpdir, engine):
+@pytest.mark.parametrize("partition_on", ("b", None))
```

Comment:
It's actually testing a bug that I uncovered while doing this (cf. the changes in `fastparquet.py`). This test will fail in `main`

Reply:
To elaborate a bit: spark writes some additional files to the dataset directory which was breaking partitioning inference because we were not properly filtering them out. By writing a partitioned dataset in the "write some extra files" test, we cover that case